include/exadg/structure/spatial_discretization/operators/nonlinear_operator.cpp

```diff
@@ -275,27 +282,83 @@ template<int dim, typename Number>
 void
 NonLinearOperator<dim, Number>::do_boundary_integral_continuous(
   IntegratorFace &                   integrator,
+  OperatorType const &               operator_type,
   dealii::types::boundary_id const & boundary_id) const
 {
   BoundaryType boundary_type = this->operator_data.bc->get_boundary_type(boundary_id);
 
   for(unsigned int q = 0; q < integrator.n_q_points; ++q)
   {
-    auto traction = calculate_neumann_value<dim, Number>(
-      q, integrator, boundary_type, boundary_id, this->operator_data.bc, this->time);
+    vector traction;
 
-    if(this->operator_data.pull_back_traction)
+    // integrate standard (stored) traction or exterior pressure on Robin boundaries
```

Comment:
This comment reads like Robin is "the only" boundary type. I think it holds for the present PR as a whole: We should aim at a modular code where Robin is one specific feature and where it would be very easy in the future to remove this feature again (with "easy" I mean minimal code changes). Even if we will never remove it again, this way of thinking might help for the design.